modMcf/src/org/aion/mcf/trie/JournalPruneDataSource.java

```diff
@@ -152,31 +158,35 @@ private Ref decRef(ByteArrayWrapper keyW) {
     }
 
     public synchronized void storeBlockChanges(BH header) {
-        if (!enabled) { return; }
+        if (!enabled) {
+            return;
+        }
         currentUpdates.blockHeader = header;
         blockUpdates.put(new ByteArrayWrapper(header.getHash()), currentUpdates);
         currentUpdates = new Updates();
     }
 
     public synchronized void prune(BH header) {
-        if (!enabled) { return; }
+        if (!enabled) {
+            return;
+        }
         ByteArrayWrapper blockHashW = new ByteArrayWrapper(header.getHash());
         Updates updates = blockUpdates.remove(blockHashW);
         if (updates != null) {
             for (ByteArrayWrapper insertedKey : updates.insertedKeys) {
                 decRef(insertedKey).dbRef = true;
             }
 
-            Map<byte[], byte[]> batchRemove = new HashMap<>();
+            List<byte[]> batchRemove = new ArrayList<>();
             for (ByteArrayWrapper key : updates.deletedKeys) {
                 Ref ref = refCount.get(key);
                 if (ref == null || ref.journalRefs == 0) {
-                    batchRemove.put(key.getData(), null);
+                    batchRemove.add(key.getData());
                 } else if (ref != null) {
                     ref.dbRef = false;
                 }
             }
-            src.putBatch(batchRemove);
+            src.deleteBatch(batchRemove);
```

Comment:
how many keys need to be removed during the prune? if it's a lot, use parallel-stream to shorten the processing time instead of the for loop?